src/plugins/connectionPlugin.ts

```diff
@@ -193,7 +194,7 @@ export type ConnectionFieldConfig<TypeName extends string = any, FieldName exten
    * This will cause the resulting type to be prefix'ed with the name of the type/field it is branched off of,
    * so as not to conflict with any non-extended connections.
    */
-  extendConnection?: (def: ObjectDefinitionBlock<any>) => void
+  extendConnection?: (def: ObjectDefinitionBlock<ResolverReturnTypeName<TypeName, FieldName>>) => void
```

Comment:
The local `extendConnection` used to be of type `any`, which prevented any field extension to be typed at all (root, args, etc...).

Same as above, we use the `ResolverReturnTypeName` typegen helper to find the connection type name of the field.

Reply:
(`ResolverReturnTypeName` was renamed to `FieldTypeName`)